<launch>

  <!-- Load temoto frames for vaultbot -->
  <rosparam command="load" file="$(find temoto)/config/vaultbot.yaml" />
 
  <!-- Bring up speech-to-text and text-to-speech -->
  <include file="$(find temoto)/launch/pocketsphinx.launch" />

  <!-- Bring up Leap Motion Controller -->
  <node name="leap_motion" pkg="leap_motion_controller" type="leap_motion" />

  <!-- Bring up Griffin Powermate -->
  <node name="griffin_powermate" pkg="griffin_powermate" type="griffin_powermate" />
  
  <!-- Bring up some temoto core components -->
  <node name="rviz_visual" pkg="temoto" type="rviz_visual" />
  <node name="interpret_utterance" pkg="temoto" type="interpret_utterance" />
  <node name="human_frame_broadcaster" pkg="temoto" type="human_frame_broadcaster" />
  
  <!-- MoveIt interface for right_ur5 -->
  <node name="move_robot" pkg="temoto" type="move_robot">
    <param name="movegroup" value="right_ur5" />
  </node>
  
  <!-- Navigation interface for move_base -->
  <node name="navigate_robot" pkg="temoto" type="navigate_robot" />

  <!-- Start temoto teleoperation -->
  <node name="start_teleop" pkg="temoto" type="start_teleop" launch-prefix="xterm -title '手元 TELEOPERATOR' -e">
    <param name="primary_hand" type="str" value="left" />
    <param name="navigate" type="bool" value="true" />
    <param name="manipulate" type="bool" value="true" />
  </node>

  <!-- bringup proper rviz config -->
  <include file="$(find temoto)/launch/vaultbot_moveit_rviz.launch">
    <arg name="config" value="true"/>
  </include>
  
</launch>